src/commands/deliverability/access.ts

```diff
@@ -51,20 +51,24 @@ export default class Access extends SfdxCommand {
   protected static requiresProject = false;
 
   public async run(): Promise<void> {
-    this.ux.startSpinner('Attempting to set Email Deliverability Access Level');
-    const accessLevel = this.flags.level as string;
-    const accessLevelValue = String(accessValuesMap.get(accessLevel?.toLowerCase()));
-    if (accessLevelValue !== 'undefined' && Number(accessLevelValue) > -1 && Number(accessLevelValue) < 3) {
-      const accessUrl = await this.parseOrgData(String(this.flags.user));
-      if (accessUrl === undefined) return;
-      try {
-        void this.toggleDeliverability(accessUrl, accessLevelValue);
-      } catch (error) {
-        this.stopSpinnerAndLogError(String(error));
+    return new Promise((resolve, reject) => {
```

Comment:
`async` implicitly returns a promise, so returning a promise is redundant. In order to `reject` a promise in an `async`, you can rethrow the error instead:

```
catch (error) {
        this.stopSpinnerAndLogError(String(error));
        throw error;
}
```